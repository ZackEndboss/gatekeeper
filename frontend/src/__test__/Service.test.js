import React from 'react';
import {Col} from 'react-bootstrap';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {MockStore} from './mockStore';
import {Provider} from 'react-redux';
import ConnectedService, {Gates, Service} from '../Service';

jest.mock('../Gate', () => 'Gate');

describe('Service component', () => {
    it('render', () => {
        const component = renderer.create(<Service group='group'
                                                   service='service'
                                                   environments={['environment1', 'environment2', 'environment3']}
                                                   environment_order={['environment1', 'environment2', 'environment3', 'environment4']}
                                                   openEditServiceDialog='openEditServiceDialog'
                                                   setService='setService'/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should contain gates for every environment', () => {
        const component = shallow(<Gates group='group'
                                         service='service'
                                         environments={['environment1', 'environment2', 'environment3']}
                                         environment_order={['environment1', 'environment2', 'environment3', 'environment4']}/>);
        expect(component.find(Col).length).toEqual(4);
    });

});

describe('load from state', () => {
    it('should load the state correctly', () => {
        const storeMock = MockStore({
            'gates': {
                'ftx': {
                    'gatekeeper': {
                        'test1': {},
                        'test2': {}
                    }
                }
            }
        });

        const subject = (
            <Provider store={storeMock}>
                <ConnectedService group='ftx' service='gatekeeper'/>
            </Provider>);
        const component = renderer.create(subject);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});