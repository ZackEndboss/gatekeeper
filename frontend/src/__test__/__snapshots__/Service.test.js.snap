// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service component render 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-1 col-xs-1"
    >
      service
    </div>
    <div>
      <div
        className="col-md-2 col-xs-2"
      />
      <div
        className="col-md-2 col-xs-2"
      />
      <div
        className="col-md-2 col-xs-2"
      />
    </div>
  </div>
</div>
`;

exports[`load from state should load the state correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-1 col-xs-1"
    >
      gatekeeper
    </div>
    <div>
      <div
        className="col-md-2 col-xs-2"
      />
      <div
        className="col-md-2 col-xs-2"
      />
    </div>
  </div>
</div>
`;
